test/test_cpp_api_parity.py

```diff
@@ -386,75 +418,129 @@ def test_methods(test_params):
         test_methods(test_params)
 
 
+def _compute_module_name(test_params_dict):
+    fullname = test_params_dict.get('fullname', None)
+    if fullname:
+        # NOTE: This doesn't work for some of the `wrap_functional` module tests such as "interpolate_nearest_1d",
+        # because in that case the module `interpolate` is not in `torch.nn` but rather in `torch.nn.functional`.
+        # We will fix this when we have parity tests for `torch.nn.functional` modules.
+        module_name = fullname.split('_')[0]
+    else:
+        module_name = test_params_dict.get('module_name')
+    return module_name
+
+
 def _process_test_params(test_params_dict, module_metadata, device):
-    module_name = test_params_dict.get('module_name')
+    module_name = _compute_module_name(test_params_dict)
     desc = test_params_dict.get('desc', None)
-    module_variant_name = module_name + (('_' + desc) if desc else '') + (('_' + device) if device != 'cpu' else '')
-    input_size = test_params_dict.get('input_size', None)
-    input_fn = test_params_dict.get('input_fn', None)
-    if input_size:
-        example_inputs = [torch.randn(input_size)]
-    elif input_fn:
-        example_inputs = list(input_fn())
+    module_variant_name = test_params_dict.get('fullname', module_name + (('_' + desc) if desc else ''))
+    module_variant_name += ('_' + device) if device != 'cpu' else ''
+    python_module_class = getattr(torch.nn, module_name)
+
+    test_params_dict['constructor'] = test_params_dict.get('constructor', python_module_class)
+    test = common_nn.TestBase(**test_params_dict)
+    try:
+        example_inputs = test._get_input()
+    except AssertionError:
+        raise RuntimeError("Missing `input_size`, `input_fn` or `input` for {}".format(module_variant_name))
```

Comment:
What is the error message in the AssertionError if this doesn't exist? It seems like this error message should exist inside `TestBase._get_input()`; it's `TestBase`'s responsibility to tell us if it was unable to get the inputs from a test_params_dict